src/js/angular/resources/controllers.js

```diff
@@ -65,211 +66,215 @@ resourcesCtrl.controller('ResourcesCtrl', ['$scope', '$timeout', 'MonitoringRest
-            if ($scope.error) {
-                return;
+        const processResponse = (response, dataSetter) => {
+            const data = response.data;
+            if (data) {
+                setChartData(data, dataSetter);
+
+                if ($scope.firstLoad) {
+                    $scope.firstLoad = false;
+
+                    const timer = $timeout(function() {
+                        $scope.loader = false;
+                    }, 500);
+
+                    $scope.$on('$destroy', function() {
+                        $timeout.cancel(timer);
+                    });
+                }
             }
-            MonitoringRestService.monitorCluster().then(function (response) {
-                const data = response.data;
-                if (data) {
-                    const timestamp = new Date();
+        };
+        const setChartData = (data, dataSetter) => {
+            const timestamp = new Date();
+            dataSetter(timestamp, data);
+        };
+        const setError = (errorHolder, error) => {
+            errorHolder.hasError = !!error;
+            if (errorHolder.hasError) {
+                errorHolder.message = getError(error);
+            }
+            $scope.error = Object.values($scope.monitors).some((monitor) => monitor.error.hasError);
+        };
+        const clearError = (errorHolder) => {
+            errorHolder.hasError = false;
+            errorHolder.message = '';
+            $scope.error = Object.values($scope.monitors).some((monitor) => monitor.error.hasError);
+        };
 
-                    $scope.clusterHealthChart.addData(timestamp, data);
+        $scope.monitors = [];
+        $scope.monitors.push({
+            error: {
+                hasError: false,
+                message: '',
+                retries: 0
+            },
+            fetchFn: getResourceMonitorData,
+            poll: null
+        });
+        $scope.monitors.push({
+            error: {
+                hasError: false,
+                message: '',
+                retries: 0
+            },
+            fetchFn: getClusterMonitorData,
+            poll: null
+        });
+        $scope.monitors.push({
+            error: {
+                hasError: false,
+                message: '',
+                retries: 0
+            },
+            fetchFn: getStructuresMonitorData,
+            poll: null
+        });
 
-                    if ($scope.firstLoad) {
-                        $scope.firstLoad = false;
+        $scope.monitors.push({
+            error: {
+                hasError: false,
+                message: '',
+                retries: 0
+            },
+            fetchFn: getQueryMonitor,
+            poll: null
+        });
 
-                        const timer = $timeout(function () {
-                            $scope.loader = false;
-                        }, 500);
+        $q.all($scope.monitors.map((monitor) => getData(monitor)))
```

Comment:
Right now, the monitor view has 3 tabs. When I open the view, all these requests are started even though I may never open onother tab in the view. Isn't it better to group these by tab somehow.

Reply:
I have made it this way so that you can compare different parameters withing the same time frame